packages/scene-composer/src/components/panels/SceneHierarchyPanel/SceneHierarchyPanel.tsx

```diff
@@ -21,10 +23,36 @@ const strings = defineMessages({
 const SceneHierarchyPanel = () => {
   const { formatMessage } = useIntl();
 
+  const sceneComposerId = useContext(sceneComposerIdContext);
+  const getSceneNodeByRef = useStore(sceneComposerId)((state) => state.getSceneNodeByRef);
+  const selectedSceneNodeRef = useStore(sceneComposerId)((state) => state.selectedSceneNodeRef);
+
   const [{ variation: canSearchHierarchy }] = useFeature(COMPOSER_FEATURES[COMPOSER_FEATURES.SceneHierarchySearch]);
   const [{ variation: canReorderHierarchy }] = useFeature(COMPOSER_FEATURES[COMPOSER_FEATURES.SceneHierarchyReorder]);
   const [{ variation: canSelectMultiple }] = useFeature(COMPOSER_FEATURES[COMPOSER_FEATURES.SceneHierarchyMultiSelect]);
 
+  const expandedNodes = useRef<string[]>([]);
+  /**
+   * List of nodes to expand based on the selected node
+   * If the selected node changes, update the existing list
+   * */
+  const nodesToExpand = useMemo(() => {
+    let currentNode = getSceneNodeByRef(selectedSceneNodeRef);
+    const pathToRoot: string[] = [];
+    while (currentNode) {
+      currentNode = getSceneNodeByRef(currentNode.parentRef);
+      if (currentNode && pathToRoot.indexOf(currentNode.ref) === -1) {
+        pathToRoot.push(currentNode.ref);
+      }
+    }
+    pathToRoot.forEach((node) => {
+      if (expandedNodes.current.indexOf(node) === -1) {
+        expandedNodes.current.push(node);
```

Comment:
will you get duplicates in the array if you select -> deselect -> reselect the same node?

Reply:
No.